apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx

```diff
@@ -165,6 +167,19 @@ class ProgressTableContainer extends React.Component {
     });
   }
 
+  onRow = row => {
```

Comment:
maybe i'm making too big of a deal out of this, but i really want us to be consistent about not using arrow functions except for inline. since this is a callback though, you can make it a regular function without needing to bind it, by binding it inline when you pass it in: `onRow={() => this.onRow()}`

here's an article with some rationale behind this pattern: https://www.bignerdranch.com/blog/choosing-the-best-approach-for-react-event-handlers/

Reply:
I will point out that we make pretty extensive use of this pattern of declaring arrow functions on the class. Certainly there is room to make the argument that we should move away from that pattern, but I would advocate for internal consistency above all else. We're currently using this pattern elsewhere, so we should use it here, too, up until we decide to change it for everywhere.

Reply:
point well taken, and after discussing this with Ben he suggested i hold a brown bag on it which i think it a great idea.

that said, we are not currently using the arrow pattern consistently. there's a lot of both patterns in our codebase. i like consistency, so when i first joined the team i looked through our style guide to see if it has an opinion. it turns out it does -- "don't use arrow functions in class fields"

our style guide defers to the [airbnb style guide](https://github.com/airbnb/javascript/tree/master/react#methods) for react. the article linked above is from their style guide explaining why to use inline arrow functions to close over local variables. 

the next section links to [this interesting article](https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1) to explain why not to use arrow functions in  class fields. 

i plan to open a PR to move this stuff into our top-level org-specific style guide, but in the meantime i don't feel like it's inappropriate to encourage developers to be consistent about not using the pattern since it _is_ specified in our style guide, albeit not in an obvious way. however, i find it more helpful to link to those articles for justification rather than just saying "do it this way because our style guide says so". 

if our codebase were currently consistent about using arrow functions, the first dictum of our style guide "Be consistent with surrounding code" would take precedent over the airbnb guide, which seems like the point you're making. but since we're not currently consistent, i could either a) not comment on it at all and accept that we're "consistently inconsistent", or b) encourage consistency in the direction supported by the airbnb guide and the points raised in those articles. i prefer (b).

but like i said, your point _is_ well taken, and i will bump up my prioritization of getting org buy-in to formalize this in a more obvious way.

Reply:
There is some conflicting information on the internet about how much arrow functions impact performance. I was looking at https://github.com/facebook/react/issues/9851, it looks like at Facebook that they use arrow functions and don't suggest that there are performance implications. It would be nice to decide as an org how to move forward with this! I really like arrow functions just because I think it makes the code cleaner and more readable. But obviously if there are performance issues then that matters.

Reply:
I'd be interested to see some of the examples in our codebase of places where we aren't using this pattern

Reply:
I think I've identified a source of confusion here, let me know if this is accurate. The pattern I'm advocating for is _not_ "bind all methods on a React component with fat arrow functions." I agree that would be silly and would incur unnecessary performance costs. However, we do obviously need to bind some methods in fat arrow functions whenever we're going to be passing them around as first-order functions to elements or other components for lazy invocation. The pattern I _am_ advocating for is "when we know in advance that a method is going to be used this way, bind it once with a fat arrow function in the class definition so we don't have to manually bind it in the render method each time we hand it over to something else." Note that from the perspective of performance, this pattern actually results in us declaring _fewer_ functions.

Reply:
cool. let's leave this as-is for now and take up the conversation with the larger group